share/mrdocs/addons/generator/common/partials/symbol/signature/record.hbs

```diff
@@ -17,4 +18,11 @@
     {{#unless (eq access ../defaultAccess)}} {{access}}{{/unless~}}
     {{#if isVirtual}} virtual{{/if}} {{>type/declarator type~}}
 {{/each~}}
-{{/unless}}{{#isSeeBelow}} { /* see-below */ }{{/isSeeBelow}};
\ No newline at end of file
+{{/unless}}
+{{!-- Unless we are going to show a class body ("{ ... }"), don't add a
+      semicolon if the record is final, because something like `class A
+      final;` is ill-formed (`final` can't be used on a non-defining
+      declaration). Similarly, don't show a semicolon if there are base
+      classes. --}}
+{{~#isSeeBelow}} { /* see-below */ }{{/isSeeBelow~}}
+{{#if (or isSeeBelow (and (not isFinal) (isEmpty bases)))}};{{/if}}
```

Comment:
Additionally, it's essential to have a formal explanation for why `not bases` did not work. Because in general, the inverse conditional with `#unless` shouldn't work either. That's an important question.

Reply:
But we have a problem with this point. If you look at `if_fn` in the same file, you'll see `#if` uses `isEmpty` and not `isTruthy`. Other built-in helpers should follow suit.

Reply:
My hunch is the problem has nothing to do with `if` and `unless`. The problem is the `and`, `or`, and all other logical helpers. 

They're not built-in helpers. They came from the Antora default UI bundle. The comments should have references to their implementation: https://gitlab.com/antora/antora-ui-default/-/tree/master/src/helpers

Their implementation relies on `isTruthy`, and we replicated that when porting it. But they should really depend on `isEmpty` because a condition that always returns `true` or `false` is useless.

Reply:
No. I said that:

1) All built-in helpers should already use isEmpty (not isTruthy)
2) All logical and antora helpers should use isEmpty instead of isTruthy